library/auth.inc

```diff
@@ -39,7 +47,28 @@ if (
     // Note we are purposefully keeping $_POST['clearPass'], which is needed for MFA to work. It is cleared from memory after a
     //  unsuccessful or successful login
     $passTemp = $_POST['clearPass'];
-    if (!(new AuthUtils('login'))->confirmPassword($_POST['authUser'], $passTemp)) {
+
+    $login_success = false;
+    if (
+        $GLOBALS['google_signin_enabled'] &&
+        $_POST['used_google_signin'] &&
+        $_POST['google_signin_token']
+    ) {
+        $user = AuthUtils::verifyGoogleSignIn($_POST['google_signin_token']);
+        if ($user !== false) {
+            // Make sure user is in an auth group
+            $authGroup = UserService::getAuthGroupForUser($user['username']);
+            if ($authGroup !== false) {
+                $login_success = true;
+                AuthUtils::setSessionVariables($user['username'], $user['password'], $user, $authGroup);
+            }
+        }
```

Comment:
I would fold the getAuthGroupForUser and setSessionVariables into the verifyGoogleSignIn function. Then it would just need to be:
```
$login_success = AuthUtils::verifyGoogleSignIn($_POST['google_signin_token']);
```

Reply:
this then allows good logging and ensure to proper validation (for example, also need to ensure the user is in a acl group)

Reply:
addressed